RecoTauTag/RecoTau/plugins/PFTauPrimaryVertexProducer.cc

```diff
@@ -43,7 +43,7 @@ void PFTauPrimaryVertexProducer::nonTauTracksInPV(const reco::VertexRef &thePVRe
 void
 PFTauPrimaryVertexProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
   auto desc = PFTauPrimaryVertexProducerBase::getDescriptionsBase();
-  descriptions.add("PFTauPrimaryVertexProducer", desc);
+  descriptions.add("pfTauPrimaryVertexProducer", desc);
```

Comment:
As such there will be two different files:
- RecoTauTag/RecoTau/python/PFTauPrimaryVertexProducer_cfi.py
- cfypython/.../RecoTauTag/RecoTau/pfTauPrimaryVertexProducer_cfi.py

You are not really adapting the module name to the "cms rules", because what will be used everywhere [1] is the unchanged definition of the explicitely written first file.

May I suggest not to pollute this longstanding pull request with non necessary commits, which could be more conveniently submitted as separate pull requests? Something that we asked you to confirm [2] was instead the unusufullness of commit 692e548. I don't think it is needed here, and if you confirm so we can go ahead and merge #26273, which is now kept blocked by this neverending PR...

[1]
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_10_6_X_2019-03-31-2300&_filestring=&_string=PFTauPrimaryVertexProducer

[2]
https://github.com/cms-sw/cmssw/pull/22594#issuecomment-477989850

Reply:
No problem to revert https://github.com/cms-sw/cmssw/commit/692e548b20520e841ae6581c1f13be67797a6f2e and sorry for not getting back to you earlier. I was about to ask whether it's a problem to revert it here since it will break unit tests until https://github.com/cms-sw/cmssw/pull/26273 is merged, so I was going to wait until it's merged.

Reply:
On the other comment on the module names, I interpreted Slava's comment "I think that these can be postponed" not as a "should be postponed", so I implemented them just to have everything done. Should these be reverted together with https://github.com/cms-sw/cmssw/commit/692e548b20520e841ae6581c1f13be67797a6f2e ?

Reply:
> No problem to revert [692e548](https://github.com/cms-sw/cmssw/commit/692e548b20520e841ae6581c1f13be67797a6f2e) and sorry for not getting back to you earlier. I was about to ask whether it's a problem to revert it here since it will break unit tests until #26273 is merged, so I was going to wait until it's merged.

If it breaks the unit tests, don't revert it now.

On the other hand, which units tests does break? The same issue is present in the IB, and there are no unit test breaking there because of it... Is some new test that you are including/modifying here?

Reply:
Yes exactly, it's the new test introduced here that exercises the tau reconstruction on top of MiniAOD, and which modifies existing configurations for that purpose, so it is affected by the incomplete configuration.

Reply:
> On the other comment on the module names, I interpreted Slava's comment "I think that these can be postponed" not as a "should be postponed", so I implemented them just to have everything done. Should these be reverted together with [692e548](https://github.com/cms-sw/cmssw/commit/692e548b20520e841ae6581c1f13be67797a6f2e) ?

That change is kind of useless, as it is implemented here, because the name of the module which will continue to be called in the cmssw configs is not really touched.

What I meant, anyhow, is that one should avoid to add extra stuffs in this long-living pull requests, because they are adding complexity, risk of interference and merge conflicts: and this is something that should be better avoided if you don't want to risk to delay further this PR. You can avoid that kind of risks by putting those commits in a separate PR (of course, if they do not overlap with what you are already touching here)